google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResumableStreamIteratorTest.java

```diff
@@ -163,12 +170,39 @@ public void simple() {
   }
 
   @Test
-  public void closedSpan() {
+  public void closedOTSpan() {
+    SpannerOptions.resetActiveTracingFramework();
+    SpannerOptions.enableOpenTelemetryTraces();
     Assume.assumeTrue(
         "This test is only supported on JDK11 and lower",
         JavaVersionUtil.getJavaMajorVersion() < 12);
 
-    Span span = mock(Span.class);
+    io.opentelemetry.api.trace.Span oTspan = mock(io.opentelemetry.api.trace.Span.class);
+    ISpan span = new OpenTelemetrySpan(oTspan);
+    when(oTspan.makeCurrent()).thenReturn(mock(Scope.class));
+    setInternalState(ResumableStreamIterator.class, this.resumableStreamIterator, "span", span);
+
+    ResultSetStream s1 = Mockito.mock(ResultSetStream.class);
+    Mockito.when(starter.startStream(null)).thenReturn(new ResultSetIterator(s1));
+    Mockito.when(s1.next())
+        .thenReturn(resultSet(ByteString.copyFromUtf8("r1"), "a"))
+        .thenReturn(resultSet(ByteString.copyFromUtf8("r2"), "b"))
+        .thenReturn(null);
+    assertThat(consume(resumableStreamIterator)).containsExactly("a", "b").inOrder();
+
+    resumableStreamIterator.close("closed");
+    verify(oTspan).end();
+  }
+
+  @Test
+  public void closedOCSpan() {
+    SpannerOptions.resetActiveTracingFramework();
+    SpannerOptions.enableOpenCensusTraces();
```

Comment:
Here also: We should prefer not to do this kind of set/reset in a separate test, and if we do, then it should preferably be in the other order:
1. Set the desired configuration
2. Reset to the default in a finally block

The above will ensure that we don't unintentionally leak a custom configuration to other tests. Now, OpenCensus traces are enabled for any test that follows this test. That can give unintended side effects, as much of this configuration is global.